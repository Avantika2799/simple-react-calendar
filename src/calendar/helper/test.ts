import { getDayEachDay, getDayIndexIsWeekend, getMonthWeeks } from '.'

describe('Helper functions', () => {
  describe('#getDayIndexIsWeekend', () => {
    describe('when weekday', () => {
      describe('when `index` is `Monday`', () => {
        test('returns `false`', () => {
          expect(getDayIndexIsWeekend({ dayIndex: 1, weekStartsOn: 1 })).toBe(false)
        })
      })

      describe('when `index` is `Tuesday`', () => {
        test('returns `false`', () => {
          expect(getDayIndexIsWeekend({ dayIndex: 2, weekStartsOn: 1 })).toBe(false)
        })
      })

      describe('when `index` is `Wednesday`', () => {
        test('returns `false`', () => {
          expect(getDayIndexIsWeekend({ dayIndex: 3, weekStartsOn: 1 })).toBe(false)
        })
      })

      describe('when `index` is `Thursday`', () => {
        test('returns `false`', () => {
          expect(getDayIndexIsWeekend({ dayIndex: 4, weekStartsOn: 1 })).toBe(false)
        })
      })

      describe('when `index` is `Friday`', () => {
        test('returns `false`', () => {
          expect(getDayIndexIsWeekend({ dayIndex: 5, weekStartsOn: 1 })).toBe(false)
        })
      })
    })

    describe('when weekend', () => {
      describe('when `index` is `Saturday`', () => {
        test('returns `true`', () => {
          expect(getDayIndexIsWeekend({ dayIndex: 6, weekStartsOn: 1 })).toBe(true)
        })
      })

      describe('when `index` is `Sunday`', () => {
        test('returns `true`', () => {
          expect(getDayIndexIsWeekend({ dayIndex: 0, weekStartsOn: 1 })).toBe(true)
        })
      })
    })
  })

  describe('#getDayEachDay', () => {
    test('returns array of Days', () => {
      expect(getDayEachDay({ start: new Date(2019, 10, 25), end: new Date(2019, 11, 1) })).toEqual([
        new Date(2019, 10, 25),
        new Date(2019, 10, 26),
        new Date(2019, 10, 27),
        new Date(2019, 10, 28),
        new Date(2019, 10, 29),
        new Date(2019, 10, 30),
        new Date(2019, 11, 1),
      ])
    })
  })

  describe('#getMonthWeeks', () => {
    describe('when no `minNumberOfWeeks` has been passed', () => {
      test('returns a weeks array', () => {
        expect(getMonthWeeks({ activeMonth: new Date(2019, 11, 14), weekStartsOn: 1 })).toEqual([
          { weekStart: new Date(2019, 10, 25), weekNumber: 48, weekEnd: new Date(2019, 11, 1) },
          { weekStart: new Date(2019, 11, 2), weekNumber: 49, weekEnd: new Date(2019, 11, 8) },
          { weekStart: new Date(2019, 11, 9), weekNumber: 50, weekEnd: new Date(2019, 11, 15) },
          { weekStart: new Date(2019, 11, 16), weekNumber: 51, weekEnd: new Date(2019, 11, 22) },
          { weekStart: new Date(2019, 11, 23), weekNumber: 52, weekEnd: new Date(2019, 11, 29) },
          { weekStart: new Date(2019, 11, 30), weekNumber: 1, weekEnd: new Date(2020, 0, 5) },
        ])
      })
    })

    // describe('when `minNumberOfWeeks` has been passed', () => {
    //   test('returns a weeks array', () => {
    //     expect(getMonthWeeks({activeMonth: new Date(2018, 0, 1), minNumberOfWeeks: 8, weekStartsOn: 1})).toEqual([
    //       {weekStart: '2019-10-28', weekNumber: 44},
    //       {weekStart: '2019-11-04', weekNumber: 45},
    //       {weekStart: '2019-11-11', weekNumber: 46},
    //       {weekStart: '2019-11-18', weekNumber: 47},
    //       {weekStart: '2019-11-25', weekNumber: 48},
    //       {weekStart: '2019-12-02', weekNumber: 49},
    //       {weekStart: '2019-12-09', weekNumber: 50},
    //       {weekStart: '2019-12-16', weekNumber: 51},
    //     ])
    //   })
    // })
  })
})
