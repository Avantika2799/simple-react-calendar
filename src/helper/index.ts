import formatDate from 'date-fns/format'

export const getISODate = (date: Date) => formatDate(date, 'YYYY-MM-DD')
export const getDayFormatted = (date: Date) => formatDate(date, 'D')
export const getMonthNameFormatted = (date: Date) => formatDate(date, 'MMMM YYYY')
