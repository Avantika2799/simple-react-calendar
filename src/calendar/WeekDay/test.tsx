import { cleanup, fireEvent, render, RenderResult } from '@testing-library/react'
import React from 'react'

import { getDayFormatted } from '../helper'
import WeekDay, { Props } from './WeekDay'

afterEach(cleanup)

describe('WeekDay', () => {
  let api: RenderResult, props: Props, date: Date, mockedCustomRender: any

  beforeEach(() => {
    date = new Date(2019, 10, 25)
    props = {
      blockClassName: 'test-class-name',
      date,
      getDayFormatted,
      handleOnClick: jest.fn(),
      handleOnEnter: jest.fn(),
      isCurrentMonth: true,
      isDisabled: false,
      isHighlighted: true,
      isMonthNext: false,
      isMonthPrev: false,
      isNonSelectable: false,
      ISODate: '2019-11-25',
      isSelectable: true,
      isSelected: false,
      isSelectionEnd: false,
      isSelectionStart: false,
      isToday: true,
      isWeekend: false,
      isWorkday: true,
    }
  })

  describe('action handlers', () => {
    beforeEach(() => {
      api = render(<WeekDay {...props} />)
    })

    test('invoke `#handleOnClick`', () => {
      const { getByText } = api

      fireEvent.click(getByText(/25/i))

      expect(props.handleOnClick).toHaveBeenCalledTimes(1)
      expect(props.handleOnEnter).not.toHaveBeenCalled()
    })

    test('invoke `#handleOnEnter`', () => {
      const { getByText } = api

      fireEvent.mouseEnter(getByText(/25/i))

      expect(props.handleOnClick).not.toHaveBeenCalledTimes(1)
      expect(props.handleOnEnter).toHaveBeenCalled()
    })
  })

  describe('when render prop has not been passed', () => {
    beforeEach(() => {
      api = render(<WeekDay {...props} />)
    })

    test('default render', () => {
      const { container } = api

      expect(container).toMatchSnapshot()
    })
  })

  describe('when render prop has been passed', () => {
    beforeEach(() => {
      mockedCustomRender = jest.fn(() => null)

      props = {
        ...props,
        customRender: mockedCustomRender,
      }
      api = render(<WeekDay {...props} />)
    })

    test('render prop has been called properly', () => {
      expect(props.customRender).toHaveBeenCalledTimes(1)
      expect(props.customRender).toHaveBeenCalledWith(props)
    })
  })
})
