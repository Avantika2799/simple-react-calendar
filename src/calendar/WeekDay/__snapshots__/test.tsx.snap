// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekDay when render prop has not been passed default render 1`] = `
<div>
  <div
    class="test-class-name-day_wrapper"
  >
    <button
      class="test-class-name-day is-current_month is-highlighted is-selectable is-today is-working_day"
      type="button"
      value="2019-11-25"
    >
      25
    </button>
  </div>
</div>
`;
